javascripts/discourse/widgets/category-header-widget.js.es6

```diff
@@ -35,21 +37,26 @@ export default createWidget("category-header-widget", {
   tagName: "span.discourse-category-banners",
 
   html() {
-    const path = window.location.pathname;
-    const category = this.register
-      .lookup("controller:navigation/category")
-      .get("category");
+    const router = getOwner(this).lookup("router:main");
+    const route = router.currentRoute;
+    let isCategoryTopicList;
+    let params;
 
-    if (!category) {
-      return;
+    if (route && route.params) {
+      params = route.params;
+      isCategoryTopicList = route.params.hasOwnProperty(
+        "category_slug_path_with_id"
+      );
     }
 
-    const isException = settings.exceptions
-      .split("|")
-      .filter(Boolean)
-      .includes(category.name);
-
-    if (/^\/c\//.test(path)) {
+    if (isCategoryTopicList) {
```

Comment:
```suggestion
    if (route && route.params && route.params.hasOwnProperty("category_slug_path_with_id")) {
```

I kind of feel we can just simplify the condition here to this instead of assigning params that are only used once.

Reply:
My main thought was code readability and a less monstrous conditional, but I think you're right. I'll change it.

Reply:
@tshenry TBH I think the opposite and I have to agree with what you are doing here -- we shouldn't always put things into a one liner conditional just to not assign it to a variable. I find what you did here much clearer. I've found that with a few of the reviews lately of my ruby code too.

`route && route.params && route.params.hasOwnProperty("category_slug_path_with_id")` does not immediately mean "oh that is a category topic list" to me.

Reply:
I actually think the variables complicates the readability of the code because we end up with two conditional statement to understand instead of one. Also, I don't find `isCategoryTopicList` any clearer than `route.params.hasOwnProperty("category_slug_path_with_id")` because the code is only interested in whether there is a category ID for a category slug path which we're interested in extracting. Anyway, my comment is just a suggestion for better readability. The approach here is fine :)

Reply:
Hmm I do see your point I guess. I am just saying that sometimes compressing complex conditionals into a variable which is named better is valid and helpful. Anyway, merge ahead Taylor!

Reply:
I really appreciate everyone's input here ❤️ I think there's totally valid arguments either way. Personally, I tend to lean toward making things a little more readable, especially since it seems like we avoid inline comments as a rule in the codebase. I'll go ahead and stick with the original for the time being. Thanks again everyone!